### DISCUSSION TITLE
tr

### DISCUSSION THREAD

**Discussion by DXV515** [webdev23/shelix#8] [Q&A] [answered]
[Screencast from 2024-08-30 21-53-33.webm](https://github.com/user-attachments/assets/546375af-8a57-4735-b5f7-6c33663b8967)
After launching the program, I observe the following behavior across different terminals. I don't think I did anything wrong during the installation. I don't see any errors in the shelix.log file

**Comment by webdev23** [comment, marked as answer]
It appears that the file watcher, is detecting a change in your working directory.

It could be due to another tool or script you are using that is, for example, writing a log file.
Note that each reloads is due to a new change, so make sure to locate what is changing in your project files (including sub directories).

You could checks last modified files within a directory with `ls -ltR | head `
Since it's blinking, we don't see but the last modified file name appears in the watcher pane, aside of `MODIFY`.

By default, the file_watcher does exclude the file `shelix.logs` and `/.git` hidden directory in your project directory.

Since there is no central configuration for this as of now, you will have to edit your file [`libs/watcher`](https://github.com/webdev23/shelix/blob/79fb02ce8bba7fcf386d32f00a71645b3a9f9fb0/libs/watcher#L28) and add more rules in the `--exclude` parameter of inotifywait.

Please refer to the documentation of `inotifywait` for further configuration that match your needs. 

Also, please edit your title, so it could help others with this loop issue!
